services/worker/src/worker/main.py

```diff
@@ -72,6 +87,33 @@ def heartbeat(self) -> None:
         if current_job:
             current_job.update(last_heartbeat=get_datetime())
 
+    def get_zombies(self) -> List[Job]:
+        started_jobs = Job.objects(status=Status.STARTED)
+        max_missing_heartbeats = self.app_config.worker.max_missing_heartbeats
+        heartbeat_time_interval_seconds = self.app_config.worker.heartbeat_time_interval_seconds
+        if max_missing_heartbeats >= 1:
+            return [
+                job
+                for job in started_jobs
+                if job.last_heartbeat is not None
+                and get_datetime()
+                >= pytz.UTC.localize(job.last_heartbeat)
+                + timedelta(seconds=max_missing_heartbeats * heartbeat_time_interval_seconds)
+            ]
+        else:
+            return []
+
+    def kill_zombies(self) -> None:
+        zombies = self.get_zombies()
+        Job.objects(pk__in=[zombie.pk for zombie in zombies]).update(status=Status.ERROR)
```

Comment:
We might need to updated also finished_at right? or maybe call the `finish_job` function in the queue

Reply:
I added `finished_at` since `finish_job` is used for a single job and here we may do it on lots of jobs.

Reply:
as mentioned in https://github.com/huggingface/datasets-server/pull/824#discussion_r1116047384, I would prefer to have this implementation inside queue.py, with unit tests.